package proto_ff;

// calm 自定义的fieldoption都在该文件中
import "yd_fieldoptions.proto";

enum    enBagPageType    // InvenrotySlotData.类型
{
  EN_BPT_EQUIP        = 1;    // 装备     // 100
  EN_BPT_STORAGE      = 2;    // 账号仓库 // 100
  EN_BPT_MATRIAL      = 3;    // 材料     // 100
  EN_BPT_PROP         = 4;    // 消耗品   // 100
  EN_BPT_SKIN         = 5;    // 时装     // 200
  EN_BPT_TITLE        = 6;    // 称号     // 50
  EN_BPT_GEM          = 7;    // 宝石     // 170
  EN_BPT_PARTNER_EQUIP        = 8;    // 伙伴装备     // 200
  EN_BPT_PARTNER_EQUIP_STONE        = 9;    // 伙伴装备勋章     // 200
  EN_BPT_PARTNER_TOKEN        = 10;    // 伙伴信物     // 100
  EN_BPT_BUFF_EQUIP = 11; //Buff装备     // 120
}

message GridItemBase
{
  optional uint64 index = 1;      //索引
  optional uint64 item_id = 2;    //物品ID
  optional uint64 item_num = 3;   //物品数量
  optional bool bind = 4;         //绑定状态
  optional uint32 level = 6;      //等级 预留(装备里，这个等级是玩家等级)
  optional uint64 expireTime = 7;//过期时间,0,永不过期
  optional uint32 type = 8;
}

message EquipExt
{
  repeated int32 base_attr = 1[(yd_fieldoptions.field_arysize)=10]; //基础属性
  repeated int32 star_attr = 2[(yd_fieldoptions.field_arysize)=10]; //星级属性
  repeated int32 blue_attr = 3[(yd_fieldoptions.field_arysize)=10]; //蓝星属性
  repeated int32 god_attr = 4[(yd_fieldoptions.field_arysize)=10];  //仙尊属性
  repeated int32 special_attr = 5[(yd_fieldoptions.field_arysize)=10]; //特殊属性
}

message ItemGridCSData
{
  required GridItemBase       base          = 1;
  optional EquipExt           equip_data    = 2;
}
